### PR TITLE
[plugin.video.hardstyletv] 1.0.0

### PR DESCRIPTION
**Pull request by aureliencousin**
### Description
HardstyleTV is an addon that streams Hardstyle livesets from the best festivals in the world. Qlimax, Defqon.1, Decibel, Hard Bass, etc

It's based on the youtube video plugin from [zag2me](https://github.com/zag2me)

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

File: plugin.video.hardstyletv/addon.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="plugin.video.hardstyletv"
+	name="HardstyleTV"
+	version="1.0.0"
+	provider-name="alpadev">
+  <requires>
+    <import addon="xbmc.python" version="2.14.0"/>
```
Comment (enen92) [inline plugin.video.hardstyletv/addon.xml]: please set xbmc.python version to `2.25.0` to match krypton. See: http://kodi.wiki/view/addon.xml

File: plugin.video.hardstyletv/addon.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    <import addon="script.module.addon.common" version="2.0.0"/>
+    <import addon="plugin.video.youtube" version="5.0.0"/>
+  </requires>
+  <extension point="xbmc.python.pluginsource" library="default.py">
+    <provides>video</provides>
+  </extension>
+  <extension point="xbmc.addon.metadata">
+    <summary lang="en">Hardstyle livesets Youtube video Add-on</summary>
```
Comment (enen92) [inline plugin.video.hardstyletv/addon.xml]: please change `en` to `en_gb`. It is the new default.

File: plugin.video.hardstyletv/addon.xml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    <description lang="en">HardstyleTV is an addon that streams Hardstyle livesets from the best festivals in the world. Qlimax, Defqon.1, Decibel, Hard Bass...we got all you need!</description>
+	<disclaimer lang="en">Some videos might be blocked in your country</disclaimer>
+    <platform>all</platform>
+    <source>https://github.com/aureliencousin/plugin.video.fullycharged</source>
+    <website>http://aurelien-cousin.com</website>
+	<language>en nl</language>
```
Comment (enen92) [inline plugin.video.hardstyletv/addon.xml]: this refers to media content language. Does the addon provides videos that are in english and dutch? If so, please ignore this comment. If not, please change it to the correct values.

File: plugin.video.hardstyletv/default.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+import os
+import sys
+import plugintools
+import xbmc,xbmcaddon
+from addon.common.addon import Addon
+
+addonID = 'plugin.video.hardstyletv'
+addon = Addon(addonID, sys.argv)
+local = xbmcaddon.Addon(id=addonID)
```
Comment (enen92) [inline plugin.video.hardstyletv/default.py]: please use only `xbmcaddon.Addon()`. No need for the id (only if you were accessing other plugin from yours).

File: plugin.video.hardstyletv/plugintools.py [hunk 5]
```diff
@@ -0,0 +1,582 @@
...
+LIST = "list"
+THUMBNAIL = "thumbnail"
+MOVIES = "movies"
+TV_SHOWS = "tvshows"
+SEASONS = "seasons"
+EPISODES = "episodes"
+OTHER = "other"
+
+# Suggested view codes for each type from different skins (initial list thanks to xbmcswift2 library)
+ALL_VIEW_CODES = {
```
Comment (enen92) [inline plugin.video.hardstyletv/plugintools.py]: this is not allowed in the repository anymore I'm afraid. Dev's should set the content type correctly and not force skin views in the code. Please remove the `ALL_VIEW_CODES` dictionary.

File: plugin.video.hardstyletv/plugintools.py [hunk 6]
```diff
@@ -0,0 +1,582 @@
...
+        result = matches[0]
+    except:
+        result = ""
+
+    return result
+
+def add_item( action="" , title="" , plot="" , url="" , thumbnail="" , fanart="" , show="" , episode="" , extra="", page="", info_labels = None, isPlayable = False , folder=True ):
+    _log("add_item action=["+action+"] title=["+title+"] url=["+url+"] thumbnail=["+thumbnail+"] fanart=["+fanart+"] show=["+show+"] episode=["+episode+"] extra=["+extra+"] page=["+page+"] isPlayable=["+str(isPlayable)+"] folder=["+str(folder)+"]")
+
+    listitem = xbmcgui.ListItem( title, iconImage="DefaultVideo.png", thumbnailImage=thumbnail )
```
Comment (enen92) [inline plugin.video.hardstyletv/plugintools.py]: `iconImage` and `thumbnailImage` are both deprecated. Please use `setArt` instead. Solution below:

```python
listitem = xbmcgui.ListItem( title )
listitem.setArt({"thumb":"DefaultVideo.png", "icon":"DefaultVideo.png"})
```

File: plugin.video.hardstyletv/plugintools.py [hunk 7]
```diff
@@ -0,0 +1,582 @@
...
+    listitem.setProperty('IsPlayable', 'true')
+    return xbmcplugin.setResolvedUrl(int(sys.argv[1]), True, listitem)
+
+def direct_play(url):
+    _log("direct_play ["+url+"]")
+
+    title = ""
+
+    try:
+        xlistitem = xbmcgui.ListItem( title, iconImage="DefaultVideo.png", path=url)
```
Comment (enen92) [inline plugin.video.hardstyletv/plugintools.py]: Same applies here. See comment above.

File: plugin.video.hardstyletv/plugintools.py [hunk 8]
```diff
@@ -0,0 +1,582 @@
...
+        _log("set_view content is tvshows")
+        xbmcplugin.setContent( int(sys.argv[1]) ,"tvshows" )
+    elif view_mode==SEASONS:
+        _log("set_view content is seasons")
+        xbmcplugin.setContent( int(sys.argv[1]) ,"seasons" )
+    elif view_mode==EPISODES:
+        _log("set_view content is episodes")
+        xbmcplugin.setContent( int(sys.argv[1]) ,"episodes" )
+
+    # Reads skin name
```
Comment (enen92) [inline plugin.video.hardstyletv/plugintools.py]: Please remove everything in this method below this line. Setting views is not allowed anymore in the repository.

File: plugin.video.hardstyletv/plugintools.py [hunk 9]
```diff
@@ -0,0 +1,582 @@
...
+
+f = open( os.path.join( os.path.dirname(__file__) , "addon.xml") )
+data = f.read()
+f.close()
+
+addon_id = find_single_match(data,'id="([^"]+)"')
+if addon_id=="":
+    addon_id = find_single_match(data,"id='([^']+)'")
+
+__settings__ = xbmcaddon.Addon(id=addon_id)
```
Comment (enen92) [inline plugin.video.hardstyletv/plugintools.py]: drop the id. Use only: `xbmcaddon.Addon()`

File: plugin.video.hardstyletv/plugintools.py [hunk 10]
```diff
@@ -0,0 +1,582 @@
...
+            view_code = view_codes.get(skin_name)
+            _log("set_view view code for "+view_mode+" in "+skin_name+" is "+str(view_code))
+            xbmc.executebuiltin("Container.SetViewMode("+str(view_code)+")")
+        else:
+            _log("set_view view code forced to "+str(view_code))
+            xbmc.executebuiltin("Container.SetViewMode("+str(view_code)+")")
+    except:
+        _log("Unable to find view code for view mode "+str(view_mode)+" and skin "+skin_name)
+
+f = open( os.path.join( os.path.dirname(__file__) , "addon.xml") )
```
Comment (enen92) [inline plugin.video.hardstyletv/plugintools.py]: Please remove everything till `__settings__`. You should not read the addon.xml to parse the addon id. It won't be even needed if you use `xbmcaddon.Addon() ` correctly (see comment below)

**Comment by MartijnKaijser** [comment]
@aureliencousin could you update according to the comments?

**Comment by aureliencousin** [comment]
@MartijnKaijser yup, i'm trying to get the updated version to work in Krypton (somehow kodi fails to read my addon.xml file)

**Comment by aureliencousin** [comment]
I updated the PR according to enen92's review

**Comment by enen92** [comment]
Thank you

**Pull request merged**